cpp/src/community/louvain.cu

```diff
@@ -15,25 +15,62 @@
  */
 
 #include <community/louvain.cuh>
+#include <experimental/louvain.cuh>
 
 namespace cugraph {
 
+namespace detail {
+
 template <typename vertex_t, typename edge_t, typename weight_t>
 std::pair<size_t, weight_t> louvain(raft::handle_t const &handle,
-                                    GraphCSRView<vertex_t, edge_t, weight_t> const &graph,
+                                    GraphCSRView<vertex_t, edge_t, weight_t> const &graph_view,
                                     vertex_t *clustering,
                                     size_t max_level,
                                     weight_t resolution)
 {
-  CUGRAPH_EXPECTS(graph.edge_data != nullptr,
+  CUGRAPH_EXPECTS(graph_view.edge_data != nullptr,
                   "Invalid input argument: louvain expects a weighted graph");
   CUGRAPH_EXPECTS(clustering != nullptr, "Invalid input argument: clustering is null");
 
-  Louvain<GraphCSRView<vertex_t, edge_t, weight_t>> runner(handle, graph);
+  Louvain<GraphCSRView<vertex_t, edge_t, weight_t>> runner(handle, graph_view);
+  return runner(clustering, max_level, resolution);
+}
 
+template <typename vertex_t,
+          typename edge_t,
+          typename weight_t,
+          bool store_transposed,
+          bool multi_gpu>
+std::pair<size_t, weight_t> louvain(
```

Comment:
Just a question, and why do we have both this one and the old one (https://github.com/rapidsai/cugraph/pull/1157/files#diff-d8514b5ca06cd2748169ce04e7ad85a5R25). Do you plan to eventually delete the old one?

Reply:
And if louvain works with only transposed or un-transposed matrix (I guess this expects CSR as the old function takes GraphCSRView), you can skip store_transposed from the template parameter list, and just provide the boolean value in graph_view_t.
e.g.
```
experimental::graph_view_t<vertex_t, edge_t, weight_t, false, multi_gpu> const &graph_view,
```

Reply:
The experimental version doesn't work yet (in this PR), this is an interim step to allow the old code and the new code to coexist.  Once the new version is complete and tested the old one will be deleted.